[stdlib] Make test for Sequence.map actually test a sequence

ahoppen opened the pull request:
AnySequence just forwards the call to map to its underlying storage which currently is an Array (and thus a collection). 
Using `MinimalSequence` guarantees that we are actually calling a `Sequence.map`.

ahoppen commented:
@swift-ci Please smoke test

airspeedswift commented:
`stride` might, at some point, become a collection. There are dedicated types in the std lib test suite for when you explicitly want to test a particular protocol – have a look for uses of `MinimalSequence`.

ahoppen committed:
[stdlib] Make test for Sequence.map actually test a sequence

AnySequence just forwards the call to map to its underlying storage
which used to be an Array (and thus a collection).

ahoppen commented:
Updated to use `MinimalSequence`.

@swift-ci Please smoke test

airspeedswift reviewed (approved):
LGTM!

rudkx commented:
@ahoppen `StdlibCollectionUnittest` is only built for validation tests, so this will fail for anyone who runs the plain test suite locally.
